### DISCUSSION TITLE
是否支持音频与3d的二次开发

### DISCUSSION THREAD

**Discussion by ICECHN** [Tencent/libpag#2335] [General] [answered]
目前社区版本不支持音频与3d特效，是否支持开发者进行二次开发以支持音频和3d特效？

**Comment by kevingpqi123** [comment, marked as answer]
pag 还原的是 AE 的效果，专注于动效领域，pag 文件中的音频数据都有 API 接口可以获取到，你们可以随意进行二次编辑
关于 3d 特性， AE 导出插件是不区分社区版和企业版的，如果你们想基于社区版本做开发，这里是没有任何障碍的，剩余就是开发工作，这里只是开发工作量和实现复杂度的问题

**Reply by ICECHN** [reply]
> pag 还原的是 AE 的效果，专注于动效领域，pag 文件中的音频数据都有 API 接口可以获取到，你们可以随意进行二次编辑 关于 3d 特性， AE 导出插件是不区分社区版和企业版的，如果你们想基于社区版本做开发，这里是没有任何障碍的，剩余就是开发工作，这里只是开发工作量和实现复杂度的问题

如果基于社区版本要做3d效果，有没什么思路建议或demo能提供下参考的吗？

**Reply by kevingpqi123** [reply]
3D 的特效复杂些，PAG 企业版支持了该能力，如果需要可以选择 PAG 企业版
